back-end/convert.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import os
+
+import img2pdf
+from PIL import Image
+
+log = logging.getLogger("convert")
+
+
+def change_extension(filename, extension):
+    file, _ = os.path.splitext(filename)
```

Comment:
Unless this is intended to be Python 3 specific, `file` is a builtin so this is shadowing it; maybe rename it to `file_` or `basename` or something.

Unlikely to be the cause of an actual bug in this particular case, but shadowing builtins is usually something to avoid in educational materials IMHO.